spec/unit/provider/package/tdnf_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+# Note that much of the functionality of the tdnf provider is already tested with yum provider tests,
+# as yum is the parent provider, via dnf
+
+provider_class = Puppet::Type.type(:package).provider(:tdnf)
+
+describe provider_class do
+  it_behaves_like 'RHEL package provider', provider_class, 'tdnf'
+end
```

Comment:
@Magisus I lied, there is one test we should add here :)

Let's have a simple test case that ensures TDNF is the default provider for PhotonOS, similar to how the DNF test does it:

```
  [ 22, 23, 24 ].each do |ver|
    it "should be the default provider on fedora#{ver}" do
      Facter.stubs(:value).with(:osfamily).returns(:redhat)
      Facter.stubs(:value).with(:operatingsystem).returns(:fedora)
      Facter.stubs(:value).with(:operatingsystemmajrelease).returns("#{ver}")
      expect(provider_class).to be_default
    end
  end
```

... though more closely resembling this check in the upstart provider:

```
  it "should be the default provider on Ubuntu" do
    Facter.expects(:value).with(:operatingsystem).returns("Ubuntu")
    Facter.expects(:value).with(:operatingsystemmajrelease).returns("12.04")
    expect(described_class.default?).to be_truthy
  end
```